examples/spinning_cube/src/lib.rs

```diff
@@ -1,68 +1,41 @@
 #[macro_use]
 extern crate gdnative as godot;
 
-use godot::init::{Property, PropertyHint, PropertyUsage};
-use godot::GodotString;
-
+#[derive(godot::NativeClass)]
+#[inherit(godot::MeshInstance)]
+#[user_data(godot::user_data::MutexData<RustTest>)]
+#[builder_helper{
+    builder.add_property(godot::init::Property {
+        name: "test/test_enum",
+        default: godot::GodotString::from_str("Hello"),
+        hint: godot::init::PropertyHint::Enum {
+            values: &["Hello", "World", "Testing"],
+        },
+        getter: |_: &RustTest| godot::GodotString::from_str("Hello"),
+        setter: (),
+        usage: godot::init::PropertyUsage::DEFAULT,
+    });
+    builder.add_property(godot::init::Property {
+        name: "test/test_flags",
+        default: 0,
+        hint: godot::init::PropertyHint::Flags {
+            values: &["A", "B", "C", "D"],
+        },
+        getter: |_: &RustTest| 0,
+        setter: (),
+        usage: godot::init::PropertyUsage::DEFAULT,
+    });
+}]
 struct RustTest {
     start: godot::Vector3,
     time: f32,
+    #[property(0.05)]
```

Comment:
Please make it clear in the syntax what the value is for: `#[property(default = 0.05)]`. This increases readability, and makes it easier to add more arguments or remove this argument later.

Reply:
Hey,

thank you for all the rather insightful feedback. I thought about it for a little bit and I agree that the current syntax isn't very clear about its effect, but I am not convinced your proposal allows for a better extension in the future.

Either way, I tried to implement it and am again stuck again in the syn crate type.

As it stand I currently try to parse the AttributeArgs as an `ExprAssign`. Decomposing this expression into a left and right part again seems rather convoluted and tricky to extend in the future.

If we go a named route, wouldn't it be better to set the property up like a struct? That way we could add additional named fields that could for example host `hint`s, `getter`s, `setter`s and the like.

But then again we would almost be passing the whole `gdnative::init::Property` struct... maybe it would be more reasonable to make a mirror of that struct with all fields as optional?

Reply:
> As it stand I currently try to parse the AttributeArgs as an ExprAssign. Decomposing this expression into a left and right part again seems rather convoluted and tricky to extend in the future.

I'm sorry, but I can't understand this statement. `AttributeArgs` is a type alias for `Vec<NestedMeta>`. I don't know how "parsing it as an `ExprAssign`" makes any sense at all. The pattern for name-value pairs is `NestedMeta::Meta(Meta::NameValue(MetaNameValue { path, lit, .. }))`. You don't need to "decompose the expression" yourself.

I understand that `syn` is vast and you can't possibly read everything in the docs before implementation, but please at least read the ones you're pointed to.

> If we go a named route, wouldn't it be better to set the property up like a struct?

No. List of name-value pairs is the standard in the Rust ecosystem. Take the serde rename attribute as an example: `#[serde(rename(serialize = "ser_name", deserialize = "de_name"))]`.

> make a mirror of that struct with all fields as optional?

No. You need a macro because of the field identifiers. How are you going to provide default getter/setters otherwise?